migrate/migrate.go

```diff
@@ -87,6 +87,48 @@ func List(ctx context.Context, session gocqlx.Session) ([]*Info, error) {
 	return v, nil
 }
 
+// Pending provides a listing of pending migrations.
+func Pending(ctx context.Context, session gocqlx.Session, f fs.FS) ([]*Info, error) {
+	applied, err := List(ctx, session)
+	if err != nil {
+		return nil, err
+	}
+
+	fm, err := fs.Glob(f, "*.cql")
+	if err != nil {
+		return nil, fmt.Errorf("list migrations: %w", err)
+	}
+
+	if len(applied) > len(fm) {
+		return nil, fmt.Errorf("database is ahead")
+	}
+
+	pending := make([]*Info, 0, len(fm)-len(applied))
+
+	for i := range applied {
+		if applied[i].Name != fm[i] {
+			return nil, fmt.Errorf("inconsistent migrations found, expected %q got %q at %d", applied[i].Name, fm[i], i)
+		}
+	}
+
+	for _, name := range fm[len(applied):] {
```

Comment:
It is better to just to go over `fm` and select only those that are not present in `applied` by `.Name`

Reply:
I put all the `applied`  in a set and check over the `fs` to find the difference.